docs/specification.md

```diff
@@ -311,89 +432,153 @@ available state entry is `D`:
+
+This mechanism is necessary for adoptions where a subset of developers do not
+use gittuf. When they submit changes to the main copy of the repository, they do
+not include RSL entries. Therefore, when a change is pushed to a branch by a
+non-gittuf user A, a gittuf user B can submit an RSL entry on their behalf.
+Additionally, the entry must identify the original user (TODO: cross-reference
+with signed pushes?) and include some evidence about why B thinks the change
+came from A (TODO: signed push can be evidence option?).
+
+TODO: decide if B needs to be authorized for the affected branch to sign for A.
+
+### Recovery Scenarios
+
+These scenarios are some examples where recovery is necessary.
+
+#### A change is made without an RSL entry
+
+Bob does not use gittuf and pushes to a branch. Alice notices this as her gittuf
+client detects a push to the branch without an accompanying RSL entry. She
+validates that the change came from Bob and creates an RSL entry on his behalf,
+identifying him and including information about how she verified it was him.
+Alice applies M2.
+
+#### An incorrect RSL entry is added
+
+There are several ways in which an RSL entry can be considered "incorrect". If
+an entry is malformed (structurally), Git may catch it if it's not a valid
+commit. In such instances, the push from a buggy client is rejected altogether,
+meaning other users are not exposed to the malformed commit.
+
+Invalid entries that are not rejected by Git must be caught by gittuf. Some
+examples of such invalid entries are:
+
+* RSL entry is for a non-existing Git reference
+* Commit recorded in RSL entry does not exist
+* Commit recorded in RSL entry does not match the tip of the corresponding Git
+  reference
+* RSL annotation contains references to RSL entries that do not exist or are not
+  RSL entries (i.e. the annotation points to other commits in the repository)
+
+Note that as invalid RSL entries are only created by buggy or malicious gittuf
+clients, these entries cannot be detected prior to them being pushed to the main
+repository.
+
+As correctly implemented gittuf clients verify the validity of RSL entries when
+they pull from the main repository, the user is warned if invalid entries are
+encountered. Then, the user can then use M1 to invalidate the incorrect entry.
+Other clients with the invalid entry only need to fetch the latest RSL entries
+to recover. The client that originally created the invalid entries must first
+switch to a correct implementation of gittuf and then sync the local
+repository's state with the main repository.
+
+If the main repository is also gittuf enabled, such incidents can be caught
+before other users receive the incorrect RSL entries. The repository, though,
+must not behave like a typical gittuf client. Instead, gittuf's repository
+behavior is slightly different as RSL entries are submitted before the changes
+they represent. The repository must wait to receive the full changes rather than
```

Comment:
Isn't it the case that clients will also have this problem to some extent?  I think overall, we need to consider this problem more holistically.

Reply:
Clients fetch changes to refs during the verification workflow based on what the RSL says. While they do have to wait, they're not in exactly the same situation as the repo.

I want to sketch this out for the repo some more (and in part this is based on whether the repo can be gittuf enabled or not). I think that using [atomic pushes)[https://git-scm.com/docs/git-push#Documentation/git-push.txt---no-atomic) mostly solves this problem for us.

Reply:
#66 for the repository-specific behavior.